# Assignment 1 - Refined

In [176]:
import pandas as pd

In [177]:
df = pd.read_csv(r"C:\Users\HP i5\Desktop\SE_Factory\Week 1\Pythonex\kc_house_data.csv")

In [178]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [179]:
df.shape

(21613, 21)

The df.shape allows to understand the size and structure of the dataframe at hand. 


In [180]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

ID: Unique identifier for each appartment, date when the appartment data was collected, price:target variable, the result we are studying. and the other attributes are features of the appartment that influence the price


In [181]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

The df.info method is very useful as it gives a summary of the pandas data frame, it give the number of columns (variables) and mentions each of them, it provides the number of non-null value count, in addition to the data types corresponding to each varibale (int, float, object). Moreover, it provides an estimate of the memory used by the data frame.

In [182]:
df['waterfront'].value_counts() 

waterfront
0    21450
1      163
Name: count, dtype: int64

Less than 1% of the houses have waterfront. 

In [183]:
df['floors'].unique()

array([1. , 2. , 1.5, 3. , 2.5, 3.5])

In [184]:
filtered_floors = df[df['floors'].isin([3, 3.5])]
print(filtered_floors['price'].mean())
print(filtered_floors['price'].median())


587145.9130434783
492000.0


In [185]:
filtered_floors = df[df['floors'].isin([2, 2.5])]
print(filtered_floors['price'].mean())
print(filtered_floors['price'].median())

656945.7381575815
547750.0


In [186]:
filtered_floors = df[df['floors'].isin([1, 1.5])]
print(filtered_floors['price'].mean())
print(filtered_floors['price'].median())

459942.9145353455
402500.0


In [187]:
df.nlargest(5, 'sqft_living')

id             date      price  bedrooms  bathrooms  \
12777  1225069038  20140505T000000  2280000.0         7       8.00   
7252   6762700020  20141013T000000  7700000.0         6       8.00   
3914   9808700762  20140611T000000  7060000.0         5       4.50   
9254   9208900037  20140919T000000  6890000.0         6       7.75   
8092   1924059029  20140617T000000  4670000.0         5       6.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
12777        13540    307752     3.0           0     4  ...     12   
7252         12050     27600     2.5           0     3  ...     13   
3914         10040     37325     2.0           1     2  ...     11   
9254          9890     31374     2.0           0     4  ...     13   
8092          9640     13068     1.0           1     4  ...     12   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
12777        9410           4130      1999             0    98053  47.6675   
7252         8570           3480      1910          1987    98102  47.6298   
3914         7680           2360      1940          2001    98004  47.6500   
9254         8860           1030      2001             0    98039  47.6305   
8092         4820           4820      1983          2009    98040  47.5570   

          long  sqft_living15  sqft_lot15  
12777 -121.986           4850      217800  
7252  -122.323           3940        8800  
3914  -122.214           3930       25449  
9254  -122.240           4540       42730  
8092  -122.210           3270       10454  

[5 rows x 21 columns]

In [188]:
df.nsmallest(5, 'sqft_living')

id             date     price  bedrooms  bathrooms  \
19452  3980300371  20140926T000000  142000.0         0       0.00   
15381  2856101479  20140701T000000  276000.0         1       0.75   
860    1723049033  20140620T000000  245000.0         1       0.75   
18379  1222029077  20141029T000000  265000.0         0       0.75   
4868   6896300380  20141002T000000  228000.0         0       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
19452          290     20875     1.0           0     0  ...      1   
15381          370      1801     1.0           0     0  ...      5   
860            380     15000     1.0           0     0  ...      5   
18379          384    213444     1.0           0     0  ...      4   
4868           390      5900     1.0           0     0  ...      4   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
19452         290              0      1963             0    98024  47.5308   
15381         370              0      1923             0    98117  47.6778   
860           380              0      1963             0    98168  47.4810   
18379         384              0      2003             0    98070  47.4177   
4868          390              0      1953             0    98118  47.5260   

          long  sqft_living15  sqft_lot15  
19452 -121.888           1620       22850  
15381 -122.389           1340        5000  
860   -122.323           1170       15000  
18379 -122.491           1920      224341  
4868  -122.261           2170        6000  

[5 rows x 21 columns]

It is clear for sure that the sqft_living is highly affecting the pricing of the appartments, and that is approximately by a factor of 10. It also impacts the nb of bedrooms and bathrooms inside each apartment, as the square feet of living space increases the nb of bathrooms and bedrooms increase proportionally. This indicates that higher living areas result in higher prices.


In [189]:
df.nlargest(5, 'price')

id             date      price  bedrooms  bathrooms  \
7252  6762700020  20141013T000000  7700000.0         6       8.00   
3914  9808700762  20140611T000000  7060000.0         5       4.50   
9254  9208900037  20140919T000000  6890000.0         6       7.75   
4411  2470100110  20140804T000000  5570000.0         5       5.75   
1448  8907500070  20150413T000000  5350000.0         5       5.00   

      sqft_living  sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
7252        12050     27600     2.5           0     3  ...     13        8570   
3914        10040     37325     2.0           1     2  ...     11        7680   
9254         9890     31374     2.0           0     4  ...     13        8860   
4411         9200     35069     2.0           0     0  ...     13        6200   
1448         8000     23985     2.0           0     4  ...     12        6720   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
7252           3480      1910          1987    98102  47.6298 -122.323   
3914           2360      1940          2001    98004  47.6500 -122.214   
9254           1030      2001             0    98039  47.6305 -122.240   
4411           3000      2001             0    98039  47.6289 -122.233   
1448           1280      2009             0    98004  47.6232 -122.220   

      sqft_living15  sqft_lot15  
7252           3940        8800  
3914           3930       25449  
9254           4540       42730  
4411           3560       24345  
1448           4600       21750  

[5 rows x 21 columns]

In [190]:
df.nsmallest(5, 'price')

id             date    price  bedrooms  bathrooms  sqft_living  \
1149   3421079032  20150217T000000  75000.0         1       0.00          670   
15293    40000362  20140506T000000  78000.0         2       1.00          780   
465    8658300340  20140523T000000  80000.0         1       0.75          430   
16198  3028200080  20150324T000000  81000.0         2       1.00          730   
8274   3883800011  20141105T000000  82000.0         3       1.00          860   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
1149      43377     1.0           0     0  ...      3         670   
15293     16344     1.0           0     0  ...      5         780   
465        5050     1.0           0     0  ...      4         430   
16198      9975     1.0           0     0  ...      5         730   
8274      10426     1.0           0     0  ...      6         860   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
1149               0      1966             0    98022  47.2638 -121.906   
15293              0      1942             0    98168  47.4739 -122.280   
465                0      1912             0    98014  47.6499 -121.909   
16198              0      1943             0    98168  47.4808 -122.315   
8274               0      1954             0    98146  47.4987 -122.341   

       sqft_living15  sqft_lot15  
1149            1160       42882  
15293           1700       10387  
465             1200        7500  
16198            860        9000  
8274            1140       11250  

[5 rows x 21 columns]

While comparing the highest and lowest prices, we may say that square feet of living space has the highest effect on the price on the appartment. With the increase of the square feet of living space, the number of bathrooms and bedrooms increase correspondingly. The grade the appartment is on also has an impact, higher grades have higher prices. Also, the sqaure feet of basement has an impact as well, where the appartments with 0 square feet of basement space are of the lowest prices.

In [191]:
df['condition'].value_counts()

condition
3    14031
4     5679
5     1701
2      172
1       30
Name: count, dtype: int64

In [192]:
df[['bedrooms']] #the double brackets give a more organized form, if only one bracket its gnna be less organized. 

bedrooms
0             3
1             3
2             2
3             4
4             3
...         ...
21608         3
21609         4
21610         2
21611         3
21612         2

[21613 rows x 1 columns]

In [193]:
df[df.bedrooms>10]

id             date     price  bedrooms  bathrooms  \
8757   1773100755  20140821T000000  520000.0        11       3.00   
15870  2402100895  20140625T000000  640000.0        33       1.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
8757          3000      4960     2.0           0     0  ...      7   
15870         1620      6000     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
8757         2400            600      1918          1999    98106  47.5560   
15870        1040            580      1947             0    98103  47.6878   

          long  sqft_living15  sqft_lot15  
8757  -122.363           1420        4960  
15870 -122.331           1330        4700  

[2 rows x 21 columns]

In [194]:
df[df.bedrooms == 33]

id             date     price  bedrooms  bathrooms  \
15870  2402100895  20140625T000000  640000.0        33       1.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15870         1620      6000     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15870        1040            580      1947             0    98103  47.6878   

          long  sqft_living15  sqft_lot15  
15870 -122.331           1330        4700  

[1 rows x 21 columns]

This number is extremely big to be in a single appartment (33 bedrooms in one apartment), so we had to check for the whole data concerning this appartment to check if it is a typo, and it is clear it is. This apartment is only 1620 sqft_living, hence not that big of living space, thus could never contain 33 bedrooms. So most obviously it is a typo.

In [195]:
df.loc[df['yr_built'] > 2000, ['price', 'yr_built']]

price  yr_built
5      1230000.0      2001
9       323000.0      2003
29      719000.0      2005
30      580500.0      2003
31      280000.0      2005
...          ...       ...
21608   360000.0      2009
21609   400000.0      2014
21610   402101.0      2009
21611   400000.0      2004
21612   325000.0      2008

[4541 rows x 2 columns]

In [196]:
df.loc[df['yr_built'] < 2000, ['price', 'yr_built']]

price  yr_built
0      221900.0      1955
1      538000.0      1951
2      180000.0      1933
3      604000.0      1965
4      510000.0      1987
...         ...       ...
21303  465000.0      1959
21332  245000.0      1920
21343  332220.0      1953
21470  249000.0      1943
21583  399950.0      1943

[16854 rows x 2 columns]

In [197]:
df.loc[df['yr_renovated'] >0, ['price', 'yr_built', 'yr_renovated']]

price  yr_built  yr_renovated
1       538000.0      1951          1991
35      696000.0      1930          2002
95      905000.0      1946          1991
103    1090000.0      1950          2010
115     740500.0      1900          1999
...          ...       ...           ...
19622  1150000.0      1940          1960
20057   434900.0      1977          2006
20444   500012.0      1962          2009
20447   356999.0      1971          2014
20962   110000.0      1968          2007

[914 rows x 3 columns]

78% of the building were built before 2000

In [198]:
avg_price_by_floors = df.groupby('floors')['price'].mean()
print(avg_price_by_floors)


floors
1.0    4.422196e+05
1.5    5.590449e+05
2.0    6.490515e+05
2.5    1.061021e+06
3.0    5.826201e+05
3.5    9.339375e+05
Name: price, dtype: float64


In [199]:
median_price_by_floors = df.groupby('floors')['price'].median()
print(median_price_by_floors)

floors
1.0    390000.0
1.5    524475.0
2.0    542950.0
2.5    799200.0
3.0    490000.0
3.5    534500.0
Name: price, dtype: float64


I couldn't find any proportionate relation between the floor level and the price. 

In [200]:
df.zipcode.unique()

array([98178, 98125, 98028, 98136, 98074, 98053, 98003, 98198, 98146,
       98038, 98007, 98115, 98107, 98126, 98019, 98103, 98002, 98133,
       98040, 98092, 98030, 98119, 98112, 98052, 98027, 98117, 98058,
       98001, 98056, 98166, 98023, 98070, 98148, 98105, 98042, 98008,
       98059, 98122, 98144, 98004, 98005, 98034, 98075, 98116, 98010,
       98118, 98199, 98032, 98045, 98102, 98077, 98108, 98168, 98177,
       98065, 98029, 98006, 98109, 98022, 98033, 98155, 98024, 98011,
       98031, 98106, 98072, 98188, 98014, 98055, 98039], dtype=int64)

This indicates that the houses are in relatively close areas.

In [201]:
df['view'].value_counts()

view
0    19489
2      963
3      510
1      332
4      319
Name: count, dtype: int64

In [202]:
df['sqft_basement'].value_counts()

sqft_basement
0      13126
600      221
700      218
500      214
800      206
       ...  
518        1
374        1
784        1
906        1
248        1
Name: count, Length: 306, dtype: int64

In [203]:
df['lat'].mean()

47.56005251931708

In [204]:
df['long'].mean()

-122.21389640494147

Each of the two mean values of latitute and longtitude are very close to most of the values in tables, with a slight decimal difference between each, meaning that the houses are found in the same area. 

In [205]:
df['grade'].value_counts()

grade
7     8981
8     6068
9     2615
6     2038
10    1134
11     399
5      242
12      90
4       29
13      13
3        3
1        1
Name: count, dtype: int64

# Assignment 2 

First of all, to consider only the houses built year 2000 and onwards, i created a new function that is a copy of the first one but includes only the houses where year built was 2000 and onwards.

In [206]:
df_2 = df.loc[df['yr_built'] >= 2000].copy()


In [207]:
df_2.head()

id             date      price  bedrooms  bathrooms  sqft_living  \
5   7237550310  20140512T000000  1230000.0         4        4.5         5420   
9   3793500160  20150312T000000   323000.0         3        2.5         1890   
29  1873100390  20150302T000000   719000.0         4        2.5         2570   
30  8562750320  20141110T000000   580500.0         3        2.5         2320   
31  2426039314  20141201T000000   280000.0         2        1.5         1190   

    sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
5     101930     1.0           0     0  ...     11        3890           1530   
9       6560     2.0           0     0  ...      7        1890              0   
29      7173     2.0           0     0  ...      8        2570              0   
30      3980     2.0           0     0  ...      8        2320              0   
31      1265     3.0           0     0  ...      7        1190              0   

    yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
5       2001             0    98053  47.6561 -122.005           4760   
9       2003             0    98038  47.3684 -122.031           2390   
29      2005             0    98052  47.7073 -122.110           2630   
30      2003             0    98027  47.5391 -122.070           2580   
31      2005             0    98133  47.7274 -122.357           1390   

    sqft_lot15  
5       101930  
9         7570  
29        6026  
30        3980  
31        1756  

[5 rows x 21 columns]

In [208]:
df_2['yr_built'].value_counts()

yr_built
2014    559
2006    454
2005    450
2004    433
2003    422
2007    417
2008    367
2001    305
2009    230
2002    222
2000    218
2013    201
2012    170
2010    143
2011    130
2015     38
Name: count, dtype: int64

In [209]:
maxprice_year = df_2.groupby(df_2['yr_built'])['price'].idxmax()
most_expensive_houses = df_2.loc[maxprice_year, ['yr_built', 'price']]
print(most_expensive_houses)


       yr_built      price
16302      2000  3710000.0
9254       2001  6890000.0
300        2002  3080000.0
12370      2003  4210000.0
2626       2004  4500000.0
19148      2005  3640000.0
16969      2006  3100000.0
18482      2007  3650000.0
4149       2008  4000000.0
1448       2009  5350000.0
20154      2010  2950000.0
7432       2011  2580000.0
10468      2012  2000000.0
21530      2013  3000000.0
5880       2014  3420000.0
21333      2015  2000000.0


In [210]:
max_bedrooms = df_2.groupby(df_2['yr_built'])['bedrooms'].idxmax()
max_bedrooms_df = df_2.loc[max_bedrooms, ['id', 'yr_built','bedrooms']]
print(max_bedrooms_df)

               id  yr_built  bedrooms
2668   4139500200      2000         6
12017   424069271      2001         7
1539   3425059222      2002         6
2852   7202331220      2003         6
18975  1778360150      2004         7
9840   7852011070      2005         6
20393  3021059304      2006         6
2345   3185600055      2007         6
13314   627300145      2008        10
20984  2149800278      2009         6
1877   1853081000      2010         5
2694   6891100260      2011         5
19793  5609000311      2012         6
8120   4006000183      2013         7
1882   2402100575      2014         6
7526   9520900210      2015         5


In [217]:
max_bathrooms = df_2.groupby(df_2['yr_built'])['bathrooms'].idxmax()
max_bathrooms_df = df_2.loc[max_bathrooms, ['id', 'yr_built','bathrooms']]
print(max_bathrooms_df)

               id  yr_built  bathrooms
6501   2626069030      2000       5.75
9254   9208900037      2001       7.75
300    3225069065      2002       5.00
12370  6065300370      2003       6.00
16773  1630700380      2004       5.75
19148  3625049042      2005       6.00
3100    622069006      2006       5.50
20578   424069279      2007       6.50
4149   6447300265      2008       5.50
1448   8907500070      2009       5.00
21506  2524069097      2010       6.50
19770  2619950060      2011       4.00
6389     42000130      2012       4.50
21148   524059063      2013       5.00
5880   2525049148      2014       5.00
20251  7853440050      2015       4.50


In [218]:
max_floors = df_2.groupby(df_2['yr_built'])['floors'].idxmax()
max_floors_df = df_2.loc[max_floors, ['id', 'yr_built','floors']]
print(max_floors_df)

               id  yr_built  floors
438    9528102772      2000     3.0
406    7821200390      2001     3.0
667    3300701615      2002     3.0
985    9528104108      2003     3.0
663    6798100662      2004     3.0
10077  1972202010      2005     3.5
14885  8673400177      2006     3.5
20308  1972200426      2007     3.5
15424  1702900664      2008     3.5
2478   4358700186      2009     3.0
116    3362400431      2010     3.0
3452   2559950110      2011     3.0
4346   1112000031      2012     3.0
1931   5379804150      2013     3.0
63     9528102996      2014     3.0
643    9385200045      2015     3.0


In [213]:
sqft_living_mean = df_2['sqft_living'].mean()
print(sqft_living_mean)

2471.24479932759


In [214]:
houses_more_sqftliving = df_2.loc[(df_2['sqft_living']> df_2['sqft_living'].mean()), ['id', 'sqft_living']]
print(houses_more_sqftliving)

               id  sqft_living
5      7237550310         5420
29     1873100390         2570
40     5547700270         2570
42     7203220400         3595
45     8035350320         3160
...           ...          ...
21598  8956200760         3118
21599  7202300110         3990
21600   249000205         4470
21605  3448900210         2520
21606  7936000429         3510

[2157 rows x 2 columns]


In [215]:
sqft_lot_mean = df_2['sqft_lot'].mean()
print(sqft_lot_mean)

12238.94788821181


In [216]:
houses_more_sqftlot = df_2.loc[(df_2['sqft_lot']> df_2['sqft_lot'].mean()), ['id', 'sqft_lot']]
print(houses_more_sqftlot)

               id  sqft_lot
5      7237550310    101930
45     8035350320     13603
79     1531000030     39683
114    8961960160     16171
147    3224079105     60984
...           ...       ...
21532   324069112     16526
21540   715010530     13000
21548  8835770330    184231
21551  1561750040     13405
21553  2522059251     15035

[460 rows x 2 columns]


In [228]:
import math 
def calc_distance(coordinate1, coordinate2):
    lat1, long1  = coordinate1 
    lat2, long2 = coordinate2 
    distance = math.acos(math.sin(lat1)*math.sin(lat2)+math.cos(lat1)*math.cos(lat2)*math.cos(long2-long1))*6371
    return distance

**Now concerning the .at and .iat methods in pandas**

The .at and .iat methods are used to access specific elements in a dataframe using either their labels (.at method) or their index(.iat method)
They are similar to the loc and iloc but are faster and more efficient

Example:

In [233]:
value_at = df.at[2, 'sqft_living']
print(value_at)

770


In [236]:
value_iat = df.iat[2, 4]  ##row then column
print(value_iat)

1.0


In [235]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

**Convert the 'date' column from object to datetime**

In [240]:
df['date'] = pd.to_datetime(df['date'])
df.head()

id       date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520 2014-10-13  221900.0         3       1.00         1180   
1  6414100192 2014-12-09  538000.0         3       2.25         2570   
2  5631500400 2015-02-25  180000.0         2       1.00          770   
3  2487200875 2014-12-09  604000.0         4       3.00         1960   
4  1954400510 2015-02-18  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]